Logging and argparse module added

CobbieCobbie opened the pull request:
Argparse:
Used to fetch $k$ and the height for the tree. Also, "--integer" or "-i" boolean sets whether its a grid or a plane

Logging:
Log dump into ${timestamp}_${k}-ary_tree_with_height_${h}.log

CobbieCobbie committed:
Logging and argparse module added

Argparse:
Used to fetch $k$ and the height for the tree. Also, "--integer" or "-i" boolean sets whether its a grid or a plane

Logging:
Log dump into ${timestamp}_${k}-ary_tree_with_height_${h}.log

CobbieCobbie committed:
PEPify and code cleanup

locojoetive commented on k-ary_tree_drawer.py:
```diff
@@ -16,32 +17,59 @@ def __repr__(self):
-            k = int(args[3])
-        if args[2] == "-h":
-            h = int(args[3])
+    timestamp = time.gmtime()
```
Das sieht chic aus.
Vielleicht kann man Zeile 67 rausschmeissen und in Zeile 112 - 115 schreiben:
```
_time = time.gmtime() - timestamp 
print(f"The process took {_time.tm_min} minutes and {_time.tm_sec} seconds.")
```
Vorausgesetzt es gibt einen Subtraktions operator overload für `time`.

locojoetive commented:
Naise Sache!
Wie aktivie/deaktiviere ich die logs?

CobbieCobbie commented:
Ah, nice, hab ich bisher nicht bedacht. Füge ich auch als flag noch hinzu, thx

CobbieCobbie committed:
Issue:

Logging not writing to a file

schlenger committed:
Fixed the logging

schlenger committed:
Updated gitignore to ignore log folder

CobbieCobbie committed:
- Delete old filename for .log files
- delete timestamp variable
- include the stdout of percentages of vertices again

CobbieCobbie committed:
Adapt README according to new flags

CobbieCobbie commented:
README adapted and code PEPified, thanks @schlenger for fixing the logging issue :) works like a charm

CobbieCobbie committed:
Rename the log file

Include the information about the maximum amount of children and the height of the corresponding tree

locojoetive reviewed (changes requested):
Ich bekomme beim Ausführen diesen Error:
`AttributeError: module 'argparse' has no attribute 'BooleanOptionalAction'`
Mache ich was falsch?

EDIT:
_The argparse.BooleanOptionalAction feature is only available in Python 3.9 and above._
https://nono.ma/attributeerror-module-argparse-has-no-attribute-booleanoptionalaction

Wenn du das so wolltest solltest du das auch in die Readme adden

locojoetive commented:
Ich würde noch:
1. Die Logging Flag in die Readme aufnehmen
2. Alle expliziten string casts durch f-strings ersetzen, oder alle f-strings durch string casts - wegen Konsistenz
